### ISSUE TITLE
Intent Recognition with Microsoft Audio Stack

### ISSUE THREAD

**Issue by alt95** [Azure-Samples/cognitive-services-speech-sdk#1891]
**Describe the bug**
I'm unable to use MAS with an IntentRecognizer with StartContinuousRecognitionAsync. I am able to use my application otherwise perfectly fine (without trying to specify audioProcessingOptions).

**To Reproduce**

```
var speechConfig = SpeechConfig.FromSubscription(_options.LuisKey, _options.LuisRegion);
var audioProcessingOptions = AudioProcessingOptions.Create(AudioProcessingConstants.AUDIO_INPUT_PROCESSING_ENABLE_DEFAULT);
var audioConfig = AudioConfig.FromStreamInput(_audioStream, audioProcessingOptions);
var recognizer = new IntentRecognizer(speechConfig, audioConfig);
```

**Expected behavior**
Intent recognizer works with MAS. 

**Version of the Cognitive Services Speech SDK**
Microsoft.Cognitive.Speech 1.22.0

**Platform, Operating System, and Programming Language**
 - OS: Linux (using FROM mcr.microsoft.com/dotnet/sdk:6.0)
 - Programming language: C#

**Additional context**
 - Caught exception processing conversation stream Exception with an error code: 0x5 (SPXERR_INVALID_ARG)
      System.ApplicationException: Exception with an error code: 0x5 (SPXERR_INVALID_ARG)
         at Microsoft.CognitiveServices.Speech.Internal.SpxExceptionThrower.ThrowIfFail(IntPtr hr)
         at Microsoft.CognitiveServices.Speech.Recognizer.FromConfig(GetRecognizerFromConfigDelegate fromConfig, SpeechConfig speechConfig, AudioConfig audioConfig)
         at Microsoft.CognitiveServices.Speech.Intent.IntentRecognizer..ctor(SpeechConfig speechConfig, AudioConfig audioConfig)
         
Might be related to this issue https://github.com/Azure-Samples/cognitive-services-speech-sdk/issues/1717

**Comment by jhakulin** [comment]
Internal work ref: 5084195

**Comment by jychoudh** [comment]
`AudioProcessingOptions.Create(int audioProcessingFlags)` can only be used when the input is from a microphone and the microphone is a microphone array. This method tries to retrieve microphone array geometry from the driver.
When you use this method with stream or file input, MAS cannot figure out the microphone array geometry required for audio processing. That's why you are getting the exception.
You will have to use either
`AudioProcessingOptions.Create(int audioProcessingFlags, PresetMicrophoneArrayGeometry microphoneArrayGeometry, SpeakerReferenceChannel speakerReferenceChannel)`
or
`AudioProcessingOptions.Create(int audioProcessingFlags, MicrophoneArrayGeometry microphoneArrayGeometry, SpeakerReferenceChannel speakerReferenceChannel)`
to provide the microphone array geometry of the microphone that was used to record the audio that you are passing as stream input.

**Comment by pankopon** [comment]
Closed as answered; the API documentation has been updated with corresponding notes and will be published with the Speech SDK 1.29.0 release.
